src/overlap_check/test/overlap_check_test.hpp

```diff
@@ -1,16 +1,19 @@
 
+#include <memory>
+
 #include "gtest/gtest.h"
+#include "moab/Core.hpp"
 
```

Comment:
Reordering includes in the cpp files forced me to move the includes into the header file.

I think in any case it was bad practice to have some of the required header includes in the cpp file (and as they got loaded before the header, it was ok...)